File: Allin.Backend.Net/Microservices/Admin/Allin.Admin.Application/Commands/BaseValues/Create/CreateBaseValueCommand.cs
using Allin.Admin.Infrastructure.Persistence;
using Allin.Common.Utilities.Mappings;
using MediatR;

namespace Allin.Admin.Application.Commands
{
    public class CreateBaseValueCommand : IRequest<bool>, IMapFrom<BaseValue, CreateBaseValueCommand>
    {
        public long BaseValueTypeId { get; set; }
        public long? ParentId { get; set; }
        public string BaseValueTitle { get; set; }
        public short Order { get; set; }
        public int Value { get; set; }
        public string Description { get; set; }
    }
}

File: allin-front-angular/src/app/subsystems/admin/models/commands/create-basevalue-command.ts

export interface CreateBaseValueCommand {
    title?: string;
    uniqueName?: string;
    description?: string;
    items?: CreateBaseValueItemCommand[];
}

export interface CreateBaseValueItemCommand {
    title?: string;
    description?: string;
    baseValueId?: number;
    parentId?: number;
    order?: number;
    value?: number;
}